lib/Doctrine/ORM/Id/FetchOneFromPrimary.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Id;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Connections\PrimaryReadReplicaConnection;
+use Doctrine\DBAL\ForwardCompatibility\Result as ForwardCompatibilityResult;
+use Doctrine\DBAL\Result;
+
+use function assert;
+
+/**
+ * @internal
+ */
+trait FetchOneFromPrimary
+{
+    /**
+     * @return string|int|false
+     */
+    private function fetchOneFromPrimary(Connection $connection, string $sql)
+    {
+        // Using `query` to force usage of the primary server in PrimaryReadReplicaConnection
+        $result = $connection instanceof PrimaryReadReplicaConnection
+            ? $connection->query($sql)
+            : $connection->executeQuery($sql);
+        if (! $result instanceof Result) {
+            ForwardCompatibilityResult::ensure($result);
```

Comment:
this must use the return value here, otherwise it is useless

Reply:
Of course, good catch. 🙈